### DISCUSSION TITLE
How to send pdf file with caption

### DISCUSSION THREAD

**Discussion by ivanwebdevs** [WhiskeySockets/Baileys#50] [General]
can someone help me to give example code to send pdf file from link please

**Comment by andresayac** [comment]
```javascript
const id = 'abcd@s.whatsapp.net' // the WhatsApp ID 


// send a document pdf! from url
const documentUrl= {
    document: {
        url: "https://github.com/pedrazadixon/sample-files/raw/main/sample_pdf.pdf"
    },
    caption: "Text caption",
    mimetype: "application/pdf"
}


await sock.sendMessage(id, documentUrl)


// send a document pdf! from local file
const documentFile = {
    document:  fs.readFileSync("sample_pdf.pdf"),
    caption: "Text caption",
    mimetype: "application/pdf"
}

await sock.sendMessage(id, documentFile)


// send a document pdf! from local file with path
const documentFilePath = {
    document: {
        url: "./sample_pdf.pdf"
    },
    caption: "Text caption",
    mimetype: "application/pdf"
}

await sock.sendMessage(id, documentFilePath)



```

**Reply by ivanwebdevs** [reply]
Thankss

**Reply by marcoscavaleiro** [reply]
THanks, but the pdf appears always like "file.pdf" how ca we pass the name of the filename?

**Reply by marcoscavaleiro** [reply]
i tried to pass the filename but didnt work..
 const pdfFile = {
        document: {
            url:data.linK_BULA,
            caption: data.LINK_BULA,
            mimetype: "application/pdf",
            filename:"teste.pdf"
        }
    };

**Reply by Rippanteq7** [reply]
> i tried to pass the filename but didnt work.. const pdfFile = { document: { url:data.linK_BULA, caption: data.LINK_BULA, mimetype: "application/pdf", filename:"teste.pdf" } };

fileName: "teste.pdf"

**Reply by marcoscavaleiro** [reply]
even with fileName didn't work...
in file messages.js
![image](https://github.com/WhiskeySockets/Baileys/assets/3036194/a69a2472-2f05-4e71-b2b1-43bd364cc23c)
the uploadData.fileName is undefined
the fileName is inside media..not uploadData
![image](https://github.com/WhiskeySockets/Baileys/assets/3036194/6908ba75-99e8-4b49-a2ea-c1544afd6c0f)

**Reply by andresayac** [reply]
![image](https://github.com/WhiskeySockets/Baileys/assets/16548795/c16d00c4-2930-4366-90ea-ad39a2f06473)
example

**Comment by Riders004** [comment]
> THanks, but the pdf appears always like "file.pdf" how ca we pass the name of the filename?



const documentFilePath = {
    document: {
        url: "./sample_pdf.pdf"
 
    caption: "Text caption",
    mimetype: "application/pdf"
},
{ 
filename: `sample_pdf.pdf`
}
}
